source/agora/test/TransactionReplacement.d

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+
+    Tests the transaction replacement logic for double spend transactions.
+
+    Copyright:
+        Copyright (c) 2019-2021 BOSAGORA Foundation
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.TransactionReplacement;
+
+version (unittest):
+
+import agora.api.Validator;
+import agora.common.Amount;
+import agora.consensus.data.genesis.Test;
+import agora.consensus.data.Transaction;
+import agora.crypto.Hash;
+import agora.crypto.Key;
+import agora.test.Base;
+
+import core.thread.osthread : Thread;
+
+///
+unittest
+{
+    TestConf conf = {
+        txs_to_nominate : 0,
+    };
+    auto network = makeTestNetwork!TestAPIManager(conf);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+
+    auto nodes = network.clients;
+    auto node0 = nodes[0];
+    auto node1 = nodes[1];
+
+    auto input_tx = GenesisBlock.txs.filter!(tx => tx.type == TxType.Payment).front();
```

Comment:
So `genesisSpendable.front()` ?

Reply:
Ah, `genesisSpendable()` gives you a range of `TxBuilder` but this needs txs. I see.